src/Physics/babylon.physicsImpostor.ts

```diff
@@ -330,13 +330,22 @@ module BABYLON {
             }
         }
 
+        /**
+         * Legacy collision detection event support
+         */
+        public onCollideEvent: (collider:BABYLON.PhysicsImpostor, collidedWith:BABYLON.PhysicsImpostor) => void = null;
+
         //event and body object due to cannon's event-based architecture.
         public onCollide = (e: { body: any }) => {
-            if (!this._onPhysicsCollideCallbacks.length) return;

```

Comment:
if (!this._onPhysicsCollideCallbacks.length) return; 

IS STILL THERE... Just moved AFTER the new onCollideEvent check... This will allow me to handle the collision check my (SHOULD I CHOOSE BY ASSIGNING onCollideEvent)
My Check in onCollideEvent is this:

`        private processPhysicsCollisions(collider: PhysicsImpostor, collidedAgainst:PhysicsImpostor) : void {
            if (collider.object != null && collidedAgainst.object != null) {
                var colliderAny:any = collider.object;
                if (colliderAny != null && colliderAny.metadata != null && colliderAny.metadata.collisionTags != null && colliderAny.metadata.collisionEvent != null) {
                    var collidedAgainstAny:any = collidedAgainst.object;
                    if ( collidedAgainstAny != null && collidedAgainstAny.metadata != null && collidedAgainstAny.metadata.tagName != null) {
                        if (colliderAny.metadata.collisionTags.length === 0 || colliderAny.metadata.collisionTags.indexOf(collidedAgainstAny.metadata.tagName) !== -1) {
                            colliderAny.metadata.collisionEvent(collidedAgainstAny);
                        }
                    }
                }
            }
        }
`

So you can see i do a form a checking that is almost like the original check for other imposter in a list... Instead i check if the otherImposter has a metadata tag that is in a list. This allows collision detection by the game object tag. Really helpful for dynamic clones created (like shots, or asteroid waves) that have a tag. Just need to detect when two rigid bodies collide with each other.

But to clarify ... Look at the WHOLE function (not just the minus in commit):

`        /**
         * Legacy collision detection event support
         */
        public onCollideEvent: (collider:BABYLON.PhysicsImpostor, collidedWith:BABYLON.PhysicsImpostor) => void = null;

        //event and body object due to cannon's event-based architecture.
        public onCollide = (e: { body: any }) => {
            var otherImpostor = this._physicsEngine.getImpostorWithPhysicsBody(e.body);
            if (otherImpostor) {
                // Legacy collision detection event support
                if (this.onCollideEvent) {
                    this.onCollideEvent(this, otherImpostor);
                }
                if (!this._onPhysicsCollideCallbacks.length) return;
                this._onPhysicsCollideCallbacks.filter((obj) => {
                    return obj.otherImpostors.indexOf(otherImpostor) !== -1
                }).forEach((obj) => {
                    obj.callback(this, otherImpostor);
                })
            }
        }
`

It basically give me a chance to handle the collision event AND STILL support the normal callback registration where i while have a list of imposters to give the registerOnPhyscisCollide...

Anyways... I really need a clean way of "Hooking" in the native collision detection and support all (and future physics engine)... Because this is the "BEST PLACE" to handle it.